general/scans/fit.py

```diff
@@ -515,7 +515,7 @@ def __init__(self):
         warnings.simplefilter("ignore", RuntimeWarning)
 
     def fit(self, x, y, err):
-        if not (x and y.size and err.size):
+        if x is None or y is None or err is None or len(x) < 1 or len(y) < 1 or len(err) < 1:
```

Comment:
I think this line could be a little shorter if we  do `if not all([x, y, z]):`

Reply:
I struggled to find a neater way of doing this because we're not sure whether the x,y,err inputs are arrays or not. If either of them are arrays, then all() and any() fall over. I've added a test for this.

Reply:
Makes sense.